tools/authoring-decisions.adoc

```diff
@@ -273,8 +273,72 @@ Each node has a minimal width and height, trying to resize it to a smaller size
+
+Now, as we understand Decision Node better, lets use the `taxRate.dmn` model as starting point to explain better how to wrap and structure the logic in it.
+
+As a first step, move `compute Payroll` node out of the `rate` node and remove its connection with the `taxRate` node.
+We will use this node later to invoke the logic wrapped in the Decision Service node.
+We can also change model name from `taxes` to `payroll`.
+
+Now, lets make the Decision Service node `rate` bigger as we will add more content to it.
+We will introduce a logic similar to `taxRate` node for computing vacation days and payment date.
+Please add two new Decision nodes into the **output** section of the `rate` node and name them as: `vacationDays` and `paymentDate`.
+Connect the existing `employee` node with both added nodes.
+Move also the existing `taxRate` node from the **encapsulated** into the **output** section.
+
+We sucessfuly changed the structure of the logic, as we moved all Decision nodes into the **output** section, the Decision Service node will return Complex data not Simple data.
+
+NOTE: See xref:adding-data-type[] to learn more about data types.
+
+Now we need to define logic for the new nodes.
+Set the `vacationDays` and `paymentDate` data type as `number`.
+We do not use `date` for the `paymentDate` because in our example the `paymentDate` is a day of the month.
+
+Both `vacationDays` and `paymentDate` will be implemented as a decision table very similar to the `taxRate` decision table.
+It is a good opportunity to use copy and paste feature we have.
+Copy the existing `taxRate` expression and paste it into `vacationDays` and `paymentDate` nodes or create the decison tables from scratch.
+
+NOTE: If you decided to use copy and paste feature, you will need to rename nodes back to `vacationDays` and `paymentDate`.
+
+Lets finish the new decision tables.
+Lets return **38** vacation days for **"US"** and **"33"** vacation days for **"UK"**.
+Lets return **15** as a payment date for **"US"** and **"10"** as a payment date for **"UK"**.
+
+For a moment, switch to the **Data types** tab.
+Add new Complex type `tPayroll` with these properties: `taxRate: number`, `vacationDays: number` and `paymentDate: number`.
+
+Switch back to the **Editor** tab.
+Rename the `rate` node to `payroll` and sets its Data type as `tPayroll`.
+Now connect the `payroll` with the `compute Payroll`.
+
+Open the `compute Payroll` node and **Reset** its expression to be an **Invocation**.
+For the Invocation we need to specify function name as `payroll` and the single parameter `employee` with passed value `employee`.
+List of paramaters for the Decision Service node can be also seen in the Properties panel **Input data** section.
+
+Then if we open **DMN Runner** and provide employee address state, we should see the payroll computed accordingly.
```

Comment:
Then if we open the **DMN Runner** and enter the employee's country, we should see the payroll computed accordingly.

Reply:
sure, will add it.